generator_process/actions/huggingface_hub.py

```diff
@@ -136,518 +159,64 @@ def hf_snapshot_download(
     model: str,
     token: str,
     revision: str | None = None
-) -> Generator[DownloadStatus, None, None]:
-    from filelock import FileLock
-    from huggingface_hub.constants import (
-        DEFAULT_REVISION,
-        HUGGINGFACE_HEADER_X_REPO_COMMIT,
-        HUGGINGFACE_HUB_CACHE,
-        REPO_TYPES,
-    )
-    from huggingface_hub.file_download import REGEX_COMMIT_HASH, repo_folder_name, hf_hub_url, _request_wrapper, hf_raise_for_status, logger, cached_download, build_hf_headers, get_hf_file_metadata, _cache_commit_hash_for_specific_revision, OfflineModeIsEnabled, _create_relative_symlink
-    from huggingface_hub.hf_api import HfApi
-    from huggingface_hub.utils import filter_repo_objects, validate_hf_hub_args, tqdm, logging, EntryNotFoundError, LocalEntryNotFoundError, RevisionNotFoundError
-
+):
```

Comment:
I've updated it to monkey patch `tqdm` in the `file_download` module instead of using the `tqdm_class` argument. Hopefully that argument can be improved in the future to provide more granular info.